distributed/client.py

```diff
@@ -903,53 +906,84 @@ def __repr__(self):
-        )
-        if scheduler is not None:
-            text += "  <li><b>Scheduler: </b>%s</li>\n" % scheduler.address
+        if scheduler is None:
+            child_repr = """<p>No scheduler connected.</p>"""
+        elif self.cluster:
+            child_repr = f"""
+                <details>
+                <summary style="margin-bottom: 20px;"><h3 style="display: inline;">Cluster Info</h3></summary>
+                {self.cluster._repr_html_()}
+                </details>
+                """
         else:
-            text += "  <li><b>Scheduler: not connected</b></li>\n"
-
-        if info and "dashboard" in info["services"]:
-            text += (
-                "  <li><b>Dashboard: </b><a href='%(web)s' target='_blank'>%(web)s</a></li>\n"
-                % {"web": self.dashboard_link}
-            )
+            child_repr = f"""
+                <details>
+                <summary style="margin-bottom: 20px;"><h3 style="display: inline;">Scheduler Info</h3></summary>
+                {info._repr_html_()}
+                </details>
+                """
+
+        client_status = ""
+
+        if not self.cluster and not self.scheduler_file:
+            client_status += """
+                <tr>
+                    <td style="text-align: left;"><strong>Connection method:</strong> Direct</td>
+                    <td style="text-align: left;"></td>
+                </tr>
+                """
 
-        text += "</ul>\n"
-
-        if info:
-            workers = list(info["workers"].values())
-            cores = sum(w["nthreads"] for w in workers)
-            memory = [w["memory_limit"] for w in workers]
-            memory = format_bytes(sum(memory)) if all(memory) else ""
-
-            text2 = (
-                '<h3 style="text-align: left;">Cluster</h3>\n'
-                '<ul style="text-align: left; list-style:none; margin: 0; padding: 0;">\n'
-                "  <li><b>Workers: </b>%d</li>\n"
-                "  <li><b>Cores: </b>%d</li>\n"
-                "  <li><b>Memory: </b>%s</li>\n"
-                "</ul>\n"
-            ) % (len(workers), cores, memory)
-
-            return (
-                '<table style="border: 2px solid white;">\n'
-                "<tr>\n"
-                '<td style="vertical-align: top; border: 0px solid white">\n%s</td>\n'
-                '<td style="vertical-align: top; border: 0px solid white">\n%s</td>\n'
-                "</tr>\n</table>"
-            ) % (text, text2)
+        if self.cluster:
+            client_status += f"""
+                <tr>
+                    <td style="text-align: left;"><strong>Connection method:</strong> Cluster object</td>
+                    <td style="text-align: left;"><strong>Cluster type:</strong> {type(self.cluster).__name__}</td>
+                </tr>
+                """
 
-        else:
-            return text
+        if self.scheduler_file:
+            client_status += f"""
```

Comment:
As it stands now if a user, who has a `Client` which is backed by a cluster manager, calls [`Client.write_scheduler_file`](https://github.com/dask/distributed/blob/0b50054e6936dc49a1a42463acc31499a15fb907/distributed/client.py#L3499) then the HTML repr will display two different entries for `Connection method:`, which seems surprising to me.

<img width="810" alt="Screen Shot 2021-06-14 at 3 37 03 PM" src="https://user-images.githubusercontent.com/11656932/121956338-a1af3e80-cd26-11eb-8a93-c37c5359a6e2.png">

What do you think about modifying the logic here to be something like:

```python
if self.cluster:
    # Code which is currently in this block
    ...
elif self.scheduler_file:
    # Code which is currently in the `if self.scheduler_file` block
    ...
else:
    # Code that is currently in the `if not self.cluster and not self.scheduler_file:` block
    ...
```

Reply:
Ah I hadn't considered that both could be set. I'll get that updated.